presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -561,6 +549,86 @@ private void invokeNoMoreSplitsIfNecessary()
         return COMPLETED_FUTURE;
     }
 
+    private Iterator<InternalHiveSplit> buildManifestFileIterator(Table table,
+            FileStatus symlinkFile,
+            FileSystem fileSystem,
+            DirectoryLister directoryLister,
+            NamenodeStats namenodeStats,
+            InternalHiveSplitFactory splitFactory,
+            boolean splittable)
+    {
+        List<Iterator<LocatedFileStatus>> fileStatusIterators = new ArrayList<>();
+        List<Path> paths = new ArrayList<>();
+
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(
+                hdfsEnvironment.doAs(hdfsContext.getIdentity().getUser(), () -> fileSystem.open(symlinkFile.getPath())), StandardCharsets.UTF_8))) {
+            CharStreams.readLines(reader).stream()
```

Comment:
could symlink be path to directory (and not file)? I think it can

Reply:
It can. I'm now falling back to the original if the list contains a directory.